### DISCUSSION TITLE
More Array Formulas

### DISCUSSION THREAD

**Discussion by BrianHung** [handsontable/hyperformula#1257] [Feature requests]
Hi, I'm considering `useArrayArithmetic: true` and one of the things I noticed is that there aren't that many array formulas: https://hyperformula.handsontable.com/guide/built-in-functions.html#array-manipulation

I looked up a [list of array formulas that excel](https://www.ablebits.com/office-addins-blog/excel-dynamic-arrays-functions-formulas/) has

[UNIQUE](https://www.ablebits.com/office-addins-blog/excel-unique-function-find-unique-values/) - extracts unique items from a range of cells.
[FILTER](https://www.ablebits.com/office-addins-blog/excel-filter-function/) - filters data based in the criteria you define.
[SORT](https://www.ablebits.com/office-addins-blog/excel-sort-function-auto-sort-data-formula/) - sorts a range of cells by a specified column.
[SORTBY](https://www.ablebits.com/office-addins-blog/excel-sortby-function-custom-sort-formula/) - sorts a range of cells by another range or array.
[RANDARRAY](https://www.ablebits.com/office-addins-blog/excel-randarray-function-generate-random-numbers/) - generates an array of random numbers.
[SEQUENCE](https://www.ablebits.com/office-addins-blog/sequence-function-excel-auto-generate-number-series/) - generates a list of sequential numbers.
[TEXTSPLIT](https://www.ablebits.com/office-addins-blog/textsplit-function-excel-split-cells-strings/) - splits strings by a specified delimiter across columns or/and rows.
[TOCOL](https://www.ablebits.com/office-addins-blog/excel-tocol-function-convert-range-to-column/) - convert an array or range to a single column.
[TOROW](https://www.ablebits.com/office-addins-blog/excel-torow-function/) - transform a range or array into a single row.
[WRAPCOLS](https://www.ablebits.com/office-addins-blog/excel-wrapcols-wraprows-functions/#wrapcols) - converts a row or column into a 2D array based on the specified number of values per row.
[WRAPROWS](https://www.ablebits.com/office-addins-blog/excel-wrapcols-wraprows-functions/#wraprows) - re-shapes a row or column into a 2D array based on the specified number of values per column.
[TAKE](https://www.ablebits.com/office-addins-blog/excel-take-function/) - extracts a specified number of contiguous rows or columns from the start or end of an array.
[DROP](https://www.ablebits.com/office-addins-blog/excel-drop-function/) - removes a certain number of rows or columns from an array.
[EXPAND](https://www.ablebits.com/office-addins-blog/excel-expand-function/) - grow an array to the specified number of rows and columns.
[CHOOSECOLS](https://www.ablebits.com/office-addins-blog/excel-choosecols-function/) - returns the specified columns from an array.
[CHOOSEROWS](https://www.ablebits.com/office-addins-blog/excel-chooserows-function/) - extracts the specified rows from an array.

Are there any plans on implementing these in the near term? If not, I might try to implement them as custom plugins and then submit a PR. Thanks!

**Comment by adrianszymanski89** [comment]
Hi @BrianHung 

Currently, we don't have plans on implementing those formulas, but if you would like to prepare a PR, that would be great.
